### DISCUSSION TITLE
Download release is slow for private repo

### DISCUSSION THREAD

**Discussion by christrt9** [community/community#65917] [Repositories]
### Select Topic Area

Show & Tell

### Body

I download multiple project from one server, and my project is always slow (100-250KB). 
I try multiple open source project small and big and the speed is better (1500-3000KB) 

I am having this issue also with curl request 

Any idea what is happening?

**Comment by arafajs** [comment]
- Try cloning the repo using SSH instead of HTTPS. SSH may have better throughput in some cases.
- For GitHub, check your account rate limits under settings. They limit unauthenticated requests which can affect download speeds.
- Use a GitHub API token when cloning or downloading to avoid unauthenticated rate limits. 
- Check if GitHub is having server issues or degraded performance in your region. This can temporarily affect download speeds.
- Try cloning on the GitHub desktop app or GitKraken instead of the command line. They may handle downloads differently.
- For large repos, consider using Git LFS which can improve cloning and fetching performance.
- Make sure you have Git configured to allow a high number of concurrent connections with `git config --global http.maxRequests 250`
- Check if you have any GitHub app or integration installed that may be interfering with API request speeds.
- Contact GitHub support if the problem persists and appears to be limited to their servers. They may be able to troubleshoot further or lift rate limits.

**Comment by LucaWintergerst** [comment]
we are seeing similar behaviour for private repos and releases, with the same speed as mentioned in the initial comment. 
Sometimes it speeds up randomly, but especially for the past 5 hours or so it has been really slow. 
We had a similar scenario last week, but it recovered again around this time of day (30m from now to be exact) 
The rest of last week, we didn't see any slowness with the downloads, it only started again earlier today. 

I definitely feels like an issue on the GitHub side, so fingers crossed that it recovers again soon and stays that way.

**Reply by LucaWintergerst** [reply]
and as expected, the speed recovered now. This is definitely something on the GitHub side I'm afraid 😕
